### PR TITLE
Remove prefix on docker labels

### PR DESCRIPTION
**Pull request by mdeggies**
### Description
This is a follow-up / fix-forward of https://github.com/hashicorp/consul/pull/14242. The label name needs to be `version` exactly, and cannot have the prefix that is listed in each label. I removed the label prefixes and verified the docker builds are succeeding and producing the right version. 

### Testing & Reproduction steps
To test locally:

1. Clone consul, checkout this branch, run `make dev-build` && `mkdir dist/linux/amd64/` && `cp ./bin/consul dist/linux/amd64/`
2. Build the ubi image: `docker build --tag=consul:1.13.0-ubi --no-cache --target=ubi --build-arg PRODUCT_VERSION=1.13.1 --build-arg BIN_NAME=consul .`
3. Build the default image: `docker build --tag=consul:1.13.0 --no-cache --target=default --build-arg PRODUCT_VERSION=1.13.1 --build-arg BIN_NAME=consul .`
4. Get the image ID from above built images w/ `docker image  ls | grep consul` 
5. Run `docker inspect --format='{{ index .Config.Labels "version" }}' $IMAGE_ID` with the image ID from above

A test run was also triggered in CI. The docker builds should pass before merging: https://github.com/hashicorp/consul/runs/7889524792?check_suite_focus=true

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Review by kpenfound — APPROVED** [review]

**Review by eculver — APPROVED** [review]
LGTM

**Comment by alvin-huang** [comment]
This is similar to #14204. I'll leave it up to the maintainers to see which to merge.

**Comment by mdeggies** [comment]
Closing in favor of @marianoasselborn 's PR! :) Thanks all.

**Pull request closed without merging**
